lib/api/v1/forms/keywords.go

```diff
@@ -0,0 +1,79 @@
+package forms
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"mime/multipart"
+
+	"github.com/markgravity/golang-ic/database"
+	"github.com/markgravity/golang-ic/lib/jobs"
+	"github.com/markgravity/golang-ic/lib/models"
+
+	"github.com/fatih/structs"
+)
+
+type KeywordsForm struct {
+	FileHeader *multipart.FileHeader `form:"file" binding:"required"`
+	User       *models.User
+}
+
+func (f *KeywordsForm) Save() error {
+	keywords, err := f.createKeywordsFromCSVFile()
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	keywords, err := f.readKeywordsFromCSVFile()
	if err != nil {
		return err
	}
	
	err := f.createKeywords(keywords)
	if err != nil {
		return err
	}
	
	return f.dispatchCrawlJobs(keywords)
```

- Can we separate both concerns (read CSV file // create into DB)?
- Let's also move the crawler dispatch into a function so it can get a name :)

Reply:
I think we can simply move the below code back to `Save()`, create more functions just only one line of code is quite a redundance IMO 🙏 

```go
db := database.GetDB()
  err = db.Create(&keywords).Error
  if err != nil {
  return err
}
```

Reply:
@malparty updated it here [fe36a4f](https://github.com/markgravity/golang-ic/pull/46/commits/fe36a4fa3828cb9bb5abe486a77e7dc89b71db69)